components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIProvider.java

```diff
@@ -256,6 +256,12 @@ public String getOpenAPIDefinition(Identifier apiId) throws APIManagementExcepti
         return super.getOpenAPIDefinition(apiId);
     }
 
+    @Override
+    public String getAsyncAPIDefinition(Identifier apiId) throws APIManagementException {
+        checkAccessControlPermission(apiId);
```

Comment:
This is not needed. accesscontrol is checked with the getAPI() call done before this

Reply:
Removed override. All APIConsumers and providers will inherit the original method from the super class.